Mobile improvements

firejake308 opened the pull request:
## Description

Makes the website semi-functional on mobile. Not sure if this fixes scrolling on iOS though

## Rationale

Applied a bunch of CSS media queries

## How to test

Open the device toolbar in Chrome DevTools and mess around with a bunch of different devices. Key ones to check include: iPhone 5/SE (the smallest), iPhone X (which a lot of users will have), iPad both horizontal and vertical, because the breakpoint is set between the two), and also regular desktop to make sure nothing broke.

## Screenshots

See #522

## Related tasks

Closes #522

gannonprudhomme commented:
Rebase this with master real quick so it doesn't fail linting anymore

gannonprudhomme commented on autoscheduler/frontend/src/components/LastUpdatedAt.tsx:
```diff
@@ -44,6 +46,18 @@ export function getLastUpdatedAtText(prev: Date, now = new Date()): string {
+    whiteSpace: 'nowrap',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    [appTheme.breakpoints.down('xs')]: {
+      display: 'none',
+    },
```
the heck is this lmao

gannonprudhomme replied:
bump

firejake308 commented:
Addressed #554. The landing page now looks better in landscape view on mobile, and users can scroll to see the remaining text. See screenshot below. Also made the "Login with Google" text wrap on extremely small screens (<348px), as discussed in #540.

|Before|After|
|--|--|
|![localhost_8000_(iPhone X) (1)](https://user-images.githubusercontent.com/10082177/116916530-11ec7f80-ac13-11eb-9c91-67ab3908db96.png)|![localhost_8000_(iPhone X)](https://user-images.githubusercontent.com/10082177/116916373-d9e53c80-ac12-11eb-9702-ac31c06add28.png)|

One **important** note is that a double scroll problem was introduced after I merged with master. #471 made it so that the cards will keep a maximum height of 500px on mobile, which means that the section rows will be scrollable, within the overall page that is also scrollable. Personally, I think this behavior is annoying but acceptable, since you have to scroll through all the sections to reach the end regardless. One workaround is to drag in the gray margins, so we could considering beefing up the margins to compensate. If we want to restore the previous behavior (where course cards get expanded all the way on mobile) I could *probably* have the JS `fixHeight()` function check `window.innerWidth` as an alnalog to CSS media queries.

gannonprudhomme commented on autoscheduler/frontend/src/components/LandingPage/SelectTerm/NavBarSelectTerm.css:
```diff
@@ -7,9 +7,16 @@
+    .select-term-button {
+        display: flex !important;
+    }
+}
```
```suggestion
}

```

gannonprudhomme replied:
bump2

gannonprudhomme reviewed (changes requested):
The problem I'm running into mainly is that (in landscape mostly) that it has a scroll bar within some div that contains most of the site, rather than the entire page be scrolling (like how portrait feels on the /schedule view). These are all in Chrome on an iPhone 11. You can see it in this video (had no idea github could show videos):

https://user-images.githubusercontent.com/7295783/117484149-8aa55180-af1b-11eb-9fba-4763efe2797d.mp4

A few other situations that look a bit off:

|Portrait not showing nav bar|Landscape not taking 100% of height|Landscape feels zoomed in a bit too much|
|--|--|--|
|![IMG_0831](https://user-images.githubusercontent.com/7295783/117484245-b0325b00-af1b-11eb-8e6d-a7381f77e54f.PNG)|![IMG_0834](https://user-images.githubusercontent.com/7295783/117484273-b88a9600-af1b-11eb-92b3-89b913234c4c.PNG)|![IMG_0832](https://user-images.githubusercontent.com/7295783/117484396-df48cc80-af1b-11eb-99a5-e1d1200b1162.PNG)|

Also note that you need to reset this with master then cherry pick these commits

firejake308 commented:
Welp, failed to replicate in Chrome on my Android device. Going to see if I can borrow my dad's iPhone tomorrow

firejake308 commented:
I believe that I've fixed the layout of the website on iPhone 6 by adding a `flex-shrink: 0;` CSS rule. I also noticed that the dropdown arrow in the Select Term button was weirdly positioned, so I added a `bottom: 9px;` to position it absolutely. This seemed to improve the layout in browser and on LambdaTest. I'd like for everyone (especially @eelyort) to try out the latest version at today's meeting, but until then, here are some screenshots for y'all until then.

|Before|After|
|--|--|
|![iOS-11 3-Firefox](https://user-images.githubusercontent.com/10082177/123554759-32364780-d747-11eb-91c9-6ec1af3cbb8c.png)|![iOS-11 3-Firefox (1)](https://user-images.githubusercontent.com/10082177/123554882-e637d280-d747-11eb-8616-3e9016264f12.png)|

firejake308 commented:
Fixed the width of the course column, as seen below:

|Before|After|
|--|--|
|![127 0 0 1_8000_schedule(iPhone 6_7_8)](https://user-images.githubusercontent.com/10082177/124678631-8a2c2700-de88-11eb-829e-6f584b9e4d5d.png)|![127 0 0 1_8000_schedule(iPhone 6_7_8) (1)](https://user-images.githubusercontent.com/10082177/124678638-8d271780-de88-11eb-9b5d-2c482cdef15d.png)|

gannonprudhomme commented on autoscheduler/frontend/src/components/App/App.tsx:
```diff
@@ -9,9 +10,15 @@ import LandingPage from '../LandingPage/LandingPage';
+
   return (
-    <div className={styles.appContainer}>
+    <Div100vh className={styles.appContainer}>
```
is div100vh necessary still? I remember it being mentioned that it didn't do anything but I could be misremembering

firejake308 replied:
I got rid of it and couldn't tell the difference, at least on my Android or Chrome DevTools. I think you're right because I never did fix the bug where you rotate your iPhone and it has the empty space at the bottom, and Div100vh was intended as a possible solution for that

firejake308 replied:
It's been a while but I think we decided it was fine as long as it looked good on iPhone 6 in portrait mode

gannonprudhomme commented on autoscheduler/frontend/src/components/App/App.tsx:
```diff
@@ -9,9 +10,15 @@ import LandingPage from '../LandingPage/LandingPage';
 
-const App: React.SFC = function App() {
+const App: React.FC = function App() {
+  React.useEffect(() => {
+    const takeFullHeight = (): void => { document.body.style.height = `${window.innerHeight}px`; };
+    window.onresize = takeFullHeight;
+    return (): void => { window.removeEventListener('resize', takeFullHeight); };
+  }, []);
```
I'd add a comment describing this briefly cause I'm mostly clueless on what this does

firejake308 replied:
I just ended up removing it because it doesn't actually help. It was basically a homemade implementation of Div100vh

gannonprudhomme reviewed (changes requested):
Just some minor questions / comments, should be good to go functionality wise

firejake308 commented:
Fixed the middle column width issue

gannonprudhomme reviewed (approved):
Confirmed in last meeting that this works as expected!

Thanks a bunch for your help Adel, you killed this as always 🚀
